tox.ini

```diff
@@ -55,6 +56,33 @@ commands =
     mypy --python-version 3.5 src/build
     mypy --python-version 2.7 src/build
 
+[testenv:py27-minversion]
+description = check minimum versions required of all dependencies
+deps =
+  importlib-metadata==0.22
+  virtualenv==20.0.35
+  typing==3.5.2.2
+  pep517==0.9.1
+  packaging==19.0
+  toml==0.10.0
+extras =
+    test
+commands =
+    pytest -rsx tests {posargs:-n auto}
+
+[testenv:{py35,py36}-minversion]
```

Comment:
IMHO we should find out what's our lowest version dependency on all versions, not just these old ones? Not/

Reply:
That would be a lot more involved, and I don't think you get nearly as much out of it (this current patch fixes a bug seen in the wild). You are unlikely to have Python 3.9 + a version of packaging that does not support 3.9 pre-installed or cached (for example). The only way that would be likely to happen if if you locked on an old version, then loaded on a new version, or if users add upper limits to library versions (Poetry's weak point, IMO but can't convince the authors otherwise) causing something to force an old version. Generally, you shouldn't have libraries older than your copy of Python sitting around.

_build_ works with the versions listed. Python may not, but build does. :) You technically only need one test, but 2.7 and 3.x are different enough that there could be a variation. I'd rather just drop the 3.6 test, and say this remains "oldest supported" (with 2.x and 3.x versions).

Reply:
You're making assumptions here that OS packaging people don't patch out our pip/wheel to older dependencies, and then users show up at our doors that the tool doesn't work. Would be nice to know for every python version what minimum dependencies must be met IMHO.

Reply:
It's not 0-gain, just much less gain for significantly more testing / time. I should probably at least run 3.9 and see how many of the deps still work. :)

Reply:
Looks like 3.9 works, so I can just expand this to everything. Doesn't take long to run.